packages/matter.js/src/cluster/Cluster.ts

```diff
@@ -23,29 +23,188 @@ export const enum AccessLevel {
 }
 
 /* Interfaces and helper methods to define a cluster attribute */
-export interface Attribute<T> { id: number, schema: TlvSchema<T>, optional: boolean, readAcl: AccessLevel, writable: boolean, scene: boolean, persistent: boolean, omitChanges: boolean, writeAcl?: AccessLevel, default?: T }
+export interface Attribute<T> {
+    id: number, schema: TlvSchema<T>,
+    optional: boolean,
+    optionalIf: number[],
```

Comment:
I would really put these just into "*Optional*" attributes -or better as own "Conditional" types. I would also define that a conditional attribute is (typing wise for this cluster) always Optional ...

And in fact "number" is the wring typing, or? I would not store the encoded values here, just the feature data

Reply:
Given that optionality is conditional, maybe optionality should be more of a query?  Something like:

```js
    isRequired(context: Cluster<lotsOfAnys>)
```

or `isOptional` works too, that just feels like more of a double negative to me for some reason.

Also maybe some of this stuff could move into an `Element` base interface/class shared by Attribute, Command and Event.  That would reduce repetition.

I do love seeing it spread out like that.  Now just needs to go into a separate file. 😉

Reply:
@lauckhart let me see if Element makes sense.

Reply:
Could do it later, just wanted to mention

Reply:
@lauckhart that didn't take long.  Adding a base Element class is a significant undertaking outside the scope of this PR.  Let's discuss this on Discord.